docs/src/main/asciidoc/datasource.adoc

```diff
@@ -405,6 +405,99 @@ AgroalDataSource usersDataSource;
 AgroalDataSource inventoryDataSource;
 ----
 
+[[datasource-active]]
+=== Activate/deactivate datasources
+
+If a datasource is configured at build time,
+by default it is active at runtime,
+that is Quarkus will start the corresponding JDBC connection pool or reactive client on application startup.
+
+To deactivate a datasource at runtime, set `quarkus.datasource[.optional name].active` to `false`.
+Then Quarkus will not start the corresponding JDBC connection pool or reactive client on application startup.
+Any attempt to use the corresponding datasource at runtime will fail with a clear error message.
+
+This is in particular useful when you want an application to be able
+to use one of a pre-determined set of datasources at runtime.
+
+[WARNING]
+====
+If another Quarkus extension relies on an inactive datasource,
+that extension might fail to start.
+
+In such case, you will need to deactivate that other extension too.
+For example see xref:hibernate-orm.adoc#persistence-unit-active[here for Hibernate ORM].
+====
+
+For example, with the following configuration:
+
+[source,properties]
+----
+quarkus.datasource."pg".db-kind=postgres
+quarkus.datasource."pg".active=false
+quarkus.datasource."pg".jdbc.url=jdbc:postgresql:///your_database
+
+quarkus.datasource."oracle".db-kind=oracle
+quarkus.datasource."oracle".active=false
+quarkus.datasource."oracle".jdbc.url=jdbc:oracle:///your_database
+----
+
+Setting `quarkus.datasource."pg".active=true` xref:config-reference.adoc#configuration-sources[at runtime]
+will make only the PostgreSQL datasource available,
+and setting `quarkus.datasource."oracle".active=true` at runtime
+will make only the Oracle datasource available.
+
+[TIP]
+====
+xref:config-reference.adoc#custom-profiles[Custom configuration profiles] can help simplify such a setup.
+By appending the following profile-specific configuration to the one above,
+you can select a persistence unit/datasource at runtime simply by passing
+xref:config-reference.adoc#multiple-profiles[setting configuration property `quarkus.profile`]:
```

Comment:
> simply by passing setting configuration property `quarkus.profile`

sounds weird?

Reply:
Right. I rephrased to:

> simply by setting `quarkus.profile`

(and I did the same for the other occurrence)